### PR TITLE
Rename Request to Inquiry

### PR DESCRIPTION
**Pull request by gbirke**
Since "Request" is commonly used in an HTTP context, rename this entity (but not the underlying table).

This addresses the comment https://github.com/wmde/FundraisingFrontend/pull/41#discussion_r49658198

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
+1

**Comment by wmde-manicki** [comment]
Name of the class has been confusing so the change is welcome. I'll make some little remarks in the diff.

Sorry for another lengthy comment but seeing this got me thinking: How about introducing clear distinction (in terms of DB structure and entity classes) between membership and other things currently stored as `Request`s (I mostly consider subscriptions here)?
For instance, how about naming this entity class something like "Membership" and use Request DB table for those, but have another entity class for subscriptions and also take a chance to add new DB table for those with a better structure and so on?

What I mean is that DB table has now number of columns that seem unused by subscription, and this `Inquiry` object is going to be used to represent different things depending on the context. Making this kind of split could make both DB a bit nicer, and (what's more important I think) would make the entity class easier to use, without taking care of all additional stuff needed by other type of usage (eg. remember to always set a zero fee for subscription - not saying that this particular things is required now), plus we could remove some evil constants from the entity class too (see for example https://github.com/wmde/FundraisingStore/pull/22).

I think this might be a time to think about such changes as:
 - you guys are doing quite some cleaning up and re-thinking of things now,
 - subscriptions are not used yet elsewhere than test/dev environments, there should be no serious implications if it is decided not to have a single entity class/DB table for both memberships and subscriptions,
 - change in this patch is already a part of a breaking change of the FundraisingStore package, so why not go further?

In the production database there are some (namely 10) "requests" that have type `other` (and not "membership") but those actually look like memberships to me (all have fees set etc). I am not really sure what those `other` things are, how did they end up in the database, etc. But this could be checked before doing some actual changes.

File: README.md [hunk 1]
```diff
@@ -44,12 +44,13 @@ run style checks use `composer cs`.
...
 
-### Version 2.0 (2016-01-13)
+### Version 2.0 (2016-mm-dd)
 
 * Rename entites. Database table names stay the same for backwards compatibility.
  * `Spenden` => `Donation`.
  * `Users` => `User`
  * `BackendImpressions` => `BackendImpression`
+ * `Request` => `Inquiry`
 * Use only English property names in public interface of `Request` and `Donation` class.
```
Comment (wmde-manicki) [inline README.md]: this sentence should then say "public interface of Inquiry" not the Request.

File: tests/unit/RequestTest.php [hunk 2]
```diff
@@ -3,31 +3,31 @@
 
 namespace WMDE\Fundraising\Store\Tests;
 
-use WMDE\Fundraising\Entities\Request;
+use WMDE\Fundraising\Entities\Inquiry;
 
 class RequestTest extends \PHPUnit_Framework_TestCase
```
Comment (wmde-manicki) [inline tests/unit/RequestTest.php]: Test class could have been renamed too.
Oh no, and I have just noticed that the bracketing is not right here (not a problem with this change, though)

**Comment by gbirke** [comment]
I accidentally included #40, don't know how to fix that.

**Comment by wmde-manicki** [comment]
Wouldn't removing other PR's commits from this PR's branch do the work? While doing that you could also consider squashing the commit that has fixed tests into the one that actually broke them, so there is no commits with failing tests in the history

**Comment by wmde-manicki** [comment]
Err, I forgot that one: so some tests (maybe one) would be failing on https://github.com/wmde/FundraisingStore/commit/a29a24e66bfd53932f28dd8da8c1d99ed569d3e1 until https://github.com/wmde/FundraisingStore/commit/34846e6bb11a4388e5886bc040adde1d1e32e7ad ?
Not a huge problem but it doesn't seem like too much effort to get that fixed and have clean commit history. So I'd rather have this done before merging.

Argh, now I see this test fixing commit cannot be simply squashed into the first renaming commit, as adding new entities also affected tests.

**Comment by wmde-manicki** [comment]
thanks for cleaning this up!

**Pull request merged**